keras/utils/audio_dataset_test.py

```diff
@@ -0,0 +1,352 @@
+        self.assertEqual(batch[1].dtype.name, "float32")
+
+    def test_audio_dataset_from_directory_validation_split(self):
+        directory = self._prepare_directory(num_classes=2, count=10)
+        dataset = audio_dataset.audio_dataset_from_directory(
+            directory,
+            batch_size=10,
+            output_sequence_length=30,
+            validation_split=0.2,
+            subset="training",
+            seed=1337,
+        )
+        batch = next(iter(dataset))
+        self.assertLen(batch, 2)
+        self.assertEqual(batch[0].shape, (8, 30, 1))
+        dataset = audio_dataset.audio_dataset_from_directory(
+            directory,
+            batch_size=10,
+            output_sequence_length=30,
+            validation_split=0.2,
+            subset="validation",
+            seed=1337,
+        )
+        batch = next(iter(dataset))
+        self.assertLen(batch, 2)
+        self.assertEqual(batch[0].shape, (2, 30, 1))
+
+    def test_audio_dataset_from_directory_manual_labels(self):
+        directory = self._prepare_directory(num_classes=2, count=2)
+        dataset = audio_dataset.audio_dataset_from_directory(
+            directory,
+            batch_size=8,
+            output_sequence_length=30,
+            labels=[0, 1],
+            shuffle=False,
+        )
+        batch = next(iter(dataset))
+        self.assertLen(batch, 2)
+        self.assertAllClose(batch[1], [0, 1])
+
+    def test_audio_dataset_from_directory_follow_links(self):
+        directory = self._prepare_directory(num_classes=2, count=25, nested_dirs=True)
+        dataset = audio_dataset.audio_dataset_from_directory(
+            directory,
+            batch_size=8,
+            output_sequence_length=30,
+            label_mode=None,
+            follow_links=True,
+        )
+        sample_count = 0
+        for batch in dataset:
+            sample_count += batch.shape[0]
+        self.assertEqual(sample_count, 25)
+
+    def test_audio_dataset_from_directory_no_audio(self):
+        directory = self._prepare_directory(num_classes=2, count=0)
+        with self.assertRaisesRegex(ValueError, "No audio found."):
+            _ = audio_dataset.audio_dataset_from_directory(directory)
+
+    def test_audio_dataset_from_directory_errors(self):
+        directory = self._prepare_directory(num_classes=3, count=5)
+
+        with self.assertRaisesRegex(
+            ValueError, "`sampling_rate` should be higher than 0. Received:"
+        ):
+            _ = audio_dataset.audio_dataset_from_directory(
+                directory, ragged=False, output_sequence_length=10, sampling_rate=-1
+            )
+
+        with self.assertRaisesRegex(
+            ValueError, "`sampling_rate` should have an integer value. Received:"
+        ):
+            _ = audio_dataset.audio_dataset_from_directory(
+                directory, ragged=False, output_sequence_length=10, sampling_rate=1.2
+            )
+
+        with self.assertRaisesRegex(
```

Comment:
This should not raise an error, right?

Reply:
Yes, sorry I forgot to remove it.

Reply:
Can you please keep the test case

```
_ = audio_dataset.audio_dataset_from_directory(
                directory, ragged=False, output_sequence_length=None
            )
```

And check the resulting sequence length?

Reply:
Also it seems the `ragged=True` case is untested, please add a test for it.

Reply:
Added.

Reply:
Thanks! Please also add the case `ragged=False, output_sequence_length=None`.

Reply:
I already added it, the test function name is `test_audio_dataset_from_directory_no_output_length_and_different_lengths` which generates audio samples with different lengths and pass it to `audio_dataset_from_directory`, and inside this test function I did not pass `ragged=False` or `output_sequence_length=None` because these are the default values.